tests/posix/common/src/pthread.c

```diff
@@ -587,12 +587,15 @@ ZTEST(posix_apis, test_pthread_descriptor_leak)
 	pthread_t pthread1;
 	pthread_attr_t attr;
 
+	zassert_ok(pthread_attr_init(&attr));
+	zassert_ok(pthread_attr_setstack(&attr, &stack_e[0][0], STACKS));
+
 	/* If we are leaking descriptors, then this loop will never complete */
 	for (size_t i = 0; i < CONFIG_MAX_PTHREAD_COUNT * 2; ++i) {
-		zassert_ok(pthread_attr_init(&attr));
-		zassert_ok(pthread_attr_setstack(&attr, &stack_e[0][0], STACKS));
 		zassert_ok(pthread_create(&pthread1, &attr, create_thread1, NULL),
 			   "unable to create thread %zu", i);
+		/* Small delay to prevent joining before the thread has been spawned */
+		k_msleep(100);
```

Comment:
I'd stick a big comment here about how this shouldn't be necessary but that there are race condition bugs in the underlying pthread and k_thread implementations.

Reply:
Will do